### PR TITLE
Introduce Octopus Server connection, not yet used

### PR DESCRIPTION
**Pull request by rain-on**
This creates an Octopus Server connection, which appears under the connections menu of a teamcity project.

This server is not used in any other feature, but lays the groundwork for future work (re-using said connection in build steps).

### CODE REVIEW

File: octopus-server/src/main/java/octopus/teamcity/server/connection/OctopusConnection.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  @Override
+  public String getType() {
+    return TYPE;
+  }
+
+  @Override
+  public String getDisplayName() {
+    return "OctopusDeploy Server";
```
Comment (long-sean) [inline octopus-server/src/main/java/octopus/teamcity/server/connection/OctopusConnection.java:44]: Do we (the company) normally stylise this as one world for display purposes?

File: octopus-server/src/main/resources/buildServerResources/editOctopusConnection.jsp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        <span class="smallNote">Provide some name to distinguish this connection from others.</span>
+    </td>
+</tr>
+<l:settingsGroup title="Server">
+    <tr>
+        <th>Octopus URL:<l:star/></th>
+        <td>
+            <props:textProperty name="${keys.serverUrlPropertyName}" className="longField"/>
+            <span class="error" id="error_${keys.serverUrlPropertyName}"></span>
+            <span class="smallNote">Specify Octopus web portal URL</span>
```
Comment (long-sean) [inline octopus-server/src/main/resources/buildServerResources/editOctopusConnection.jsp:37]: Remove "web portal"

**Review by long-sean — COMMENTED** [review]
See comments.

File: octopus-server/src/main/java/octopus/teamcity/server/connection/OctopusConnection.java (hunk 1 above)
Comment (rain-on) [reply octopus-server/src/main/java/octopus/teamcity/server/connection/OctopusConnection.java]: changed to "Octopus Deploy"

**Review by long-sean — APPROVED** [review]
LGTM

**Pull request merged**